cmd/terminus/main.go

```diff
@@ -0,0 +1,142 @@
+package main
+
+import (
+	"context"
+	dbsql "database/sql"
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"regexp"
+	"syscall"
+
+	"github.com/treeverse/terminus/pkg/http"
+	"github.com/treeverse/terminus/pkg/queue_handler"
+	"github.com/treeverse/terminus/pkg/store/sql"
+
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sqs"
+	"github.com/dustin/go-humanize"
+	_ "github.com/jackc/pgx/v4/stdlib"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+func main() {
+	Execute()
+}
+
+func DieOnErr(err error) {
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+}
+
+func NewSQS() (*sqs.SQS, error) {
+	sess, err := session.NewSessionWithOptions(session.Options{
+		SharedConfigState: session.SharedConfigEnable,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("New AWS session: %w", err)
+	}
+	sqs := sqs.New(sess)
+	return sqs, nil
+}
+
+var rootCmd = &cobra.Command{
+	Use:   "terminus",
+	Short: "Terminus monitors and optionally controls quotas for lakeFS users on S3",
+	Long: `Terminus is a service that listens to S3 events in order to track bytes used.
+It may can track S3 resources used by lakeFS installations or users.`,
+}
+
+// GetFlagBytes returns the value of flag using humanized bytes.  E.g. "8K" -> 8192.
+func GetFlagBytes(flags *pflag.FlagSet, flag string) (int64, error) {
+	s, err := flags.GetString(flag)
+	if err != nil {
+		return 0, fmt.Errorf("get flag %s: %w", flag, err)
+	}
+	bytes, err := humanize.ParseBytes(s)
+	if err != nil {
+		return 0, fmt.Errorf("parse flag %s value %s: %w", flag, s)
+	}
+	if int64(bytes) < 0 {
+		return int64(bytes), fmt.Errorf("Quota bytes %s too large", flag)
+	}
+	return int64(bytes), nil
+}
+
+var runCmd = &cobra.Command{
+	Use:     "run",
+	Short:   "Start the Terminus server",
+	Example: "terminus run --sqs-name=terminus-queue --db-dsn=postgres:/// --default-quota=1G",
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx := context.Background()
+		dbDriver, err := cmd.Flags().GetString("db-driver")
+		DieOnErr(err)
+		dbDSN, err := cmd.Flags().GetString("db-dsn")
+		DieOnErr(err)
```

Comment:
You wrote on the #5 review::

> This pattern returns multiple times. Maybe we should refactor into designated function:
> ```go
> func GetStringFromFlagsOrDie(flag string) string {
>   val, err := cmd.Flags().GetString(flag)
>   DieOnErr(err)
>   return val
> }
> ```
> 
> And then:
> ```go
> dbDriver := GetStringFromFlagsOrDie("db-driver")
> dbDsn := GetStringFromFlagsOrDie("db-dsn")
> ```
> etc

I don't know what is more Go-ish (generally things-that-die are not done, but this is a UI cmd so things are different), and @nopcoder can weigh in.  So doing it your way to speed approval.